wrappers/mongodb/mongodb.go

```diff
@@ -0,0 +1,117 @@
+}
+
+// NewCommandMonitor returns a new event.CommandMonitor for tracing commands
+func NewCommandMonitor() *event.CommandMonitor {
+	cm := commandMonitor{spans: make(map[spanKey]opentracing.Span)}
+	return &event.CommandMonitor{Started: cm.started, Succeeded: cm.succeeded, Failed: cm.failed}
+}
+
+func (c *commandMonitor) started(ctx context.Context, event *event.CommandStartedEvent) {
+	span, _ := opentracing.StartSpanFromContext(ctx, strings.ToUpper(event.CommandName))
+	rawSpan, ok := tracer.GetRaw(span)
+	if !ok {
+		return
+	}
+
+	c.Lock()
+	// Store span to use it on command finished
+	c.spans[spanKey{event.ConnectionID, event.RequestID}] = span
+	c.Unlock()
+
+	beforeCall(rawSpan, event)
+}
+
+func (c *commandMonitor) succeeded(ctx context.Context, event *event.CommandSucceededEvent) {
+	c.finished(ctx, &event.CommandFinishedEvent, "")
+}
+
+func (c *commandMonitor) failed(ctx context.Context, event *event.CommandFailedEvent) {
+	c.finished(ctx, &event.CommandFinishedEvent, event.Failure)
+}
+
+func (c *commandMonitor) finished(ctx context.Context, event *event.CommandFinishedEvent, failure string) {
+	key := spanKey{event.ConnectionID, event.RequestID}
+
+	c.Lock()
+	// Retrieve span set in command started
+	span, ok := c.spans[key]
+	if ok {
+		delete(c.spans, key)
+	}
+	c.Unlock()
+	if !ok {
+		return
+	}
+
+	if failure != "" {
+		utils.SetSpanError(span, failure)
+	}
+	span.Finish()
+}
+
+func beforeCall(span *tracer.RawSpan, event *event.CommandStartedEvent) {
+	span.ClassName = constants.ClassNames["MONGODB"]
+	span.DomainName = constants.DomainNames["DB"]
+
+	host, port := "", "27017"
+	if len(event.ConnectionID) > 0 {
+		host = strings.Split(event.ConnectionID, "[")[0]
+
+		if len(strings.Split(host, ":")) > 1 {
+			port = strings.Split(host, ":")[1]
+			host = strings.Split(host, ":")[0]
+		}
+	}
+
+	collectionValue := event.Command.Lookup(event.CommandName)
+	collectionName, _ := collectionValue.StringValueOK()
+
+	// Set span tags
+	tags := map[string]interface{}{
+		constants.SpanTags["OPERATION_TYPE"]:          constants.MongoDBCommandTypes[strings.ToUpper(event.CommandName)],
+		constants.DBTags["DB_TYPE"]:                   "mongodb",
+		constants.DBTags["DB_HOST"]:                   host,
+		constants.DBTags["DB_PORT"]:                   port,
+		constants.DBTags["DB_INSTANCE"]:               event.DatabaseName,
+		constants.MongoDBTags["MONGODB_COMMAND_NAME"]: strings.ToUpper(event.CommandName),
+		constants.MongoDBTags["MONGODB_COLLECTION"]:   collectionName,
+		constants.SpanTags["TRIGGER_DOMAIN_NAME"]:     constants.AwsLambdaApplicationDomain,
+		constants.SpanTags["TRIGGER_CLASS_NAME"]:      constants.AwsLambdaApplicationClass,
+		constants.SpanTags["TRIGGER_OPERATION_NAMES"]: []string{application.FunctionName},
+		constants.SpanTags["TOPOLOGY_VERTEX"]:         true,
+	}
+
+	if !config.MaskMongoDBCommand {
+		if event.Command != nil {
+			tags[constants.MongoDBTags["MONGODB_COMMAND"]] = event.Command.String()
```

Comment:
We had this [command size limit](https://github.com/thundra-io/thundra-lambda-agent-python/blob/17643a02e425714377c5607efa069b90dc432894/thundra/integrations/mongodb.py#L68) in the python agent. Don't we need a similar check here too?